### PR TITLE
metric control: Add the bar_color property

### LINKED ISSUE #1644: Metric control: Add the bar_color property
Adding the bar_color attribute to the Metric component allows users to change the color of the bar.

### PR DESCRIPTION
**Pull request by namnguyen20999**
Adding bar_color property to Metric

```
from taipy.gui import Gui

color_map = {0: "green", 25: "yellow", 40: "red"}


page = """
# Slider - List of values

<|30|metric|bar_color=#e6b87c|color_map={color_map}|>
"""
```

image

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 1]
```diff
@@ -109,32 +115,32 @@ const Metric = (props: MetricProps) => {
...
                 gauge: {
                     axis: {
-                        range: [
-                            props.min || 0,
-                            props.max || 100
-                        ]
+                        range: [props.min || 0, props.max || 100],
+                    },
+                    bar: {
+                        color: props.barColor,
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx:130]: 👍

**Review by FredLL-Avaiga — DISMISSED** [review]

File: doc/gui/examples/controls/chat-discuss.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 from taipy.gui.gui_actions import navigate, notify
 
 username = ""
-users: list[str|Icon] = []
+users: list[str | Icon] = []
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/controls/chat-discuss.py:27]: Is this supported in python 3.8 ?

File: taipy/gui/viselements.json [hunk 3]
```diff
@@ -51,7 +51,7 @@
                     {
                         "name": "on_action",
                         "type": "Callback",
-                        "doc": "The name of a function that is triggered when the button is pressed.<br/>The parameters of that function are all optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (optional[str]): the identifier of the button.</li>\n<li>payload (dict): a dictionary that contains the key \"action\" set to the name of the action that triggered this callback.</li>\n</ul>",
+                        "doc": "The name of a function that is triggered when the button is pressed.<br/>The parameters of that function are all optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str|None): the identifier of the button.</li>\n<li>payload (dict): a dictionary that contains the key \"action\" set to the name of the action that triggered this callback.</li>\n</ul>",
```
Comment (FredLL-Avaiga) [inline taipy/gui/viselements.json:54]: I liked the `optional`, I think it's more clear for the user

**Review by FredLL-Avaiga — DISMISSED** [review]

File: taipy/gui/viselements.json (hunk 3 above)
Comment (FabienLelaquais) [reply taipy/gui/viselements.json:54]: The problem is with the Page Builder API.
type1 | type2 is a valid Python construct that, when injected in the function signature (see https://github.com/Avaiga/taipy/issues/1632), can be interpreted.

**Review by FabienLelaquais — DISMISSED** [review]
All good!

File: taipy/gui/viselements.json (hunk 3 above)
Comment (FredLL-Avaiga) [reply taipy/gui/viselements.json:54]: not for python 3.8 ?

Comment (FabienLelaquais) [reply taipy/gui/viselements.json:54]: Bloody good point. Let's discuss that

File: doc/gui/examples/controls/chat-discuss.py (hunk 2 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/chat-discuss.py:27]: I just double-check it and it's not. @FabienLelaquais What's your thought on this

Comment (FabienLelaquais) [reply doc/gui/examples/controls/chat-discuss.py:27]: My thoughts:
all this comes from the generation of the Page Builder API, in taipy/gui/builder/\_\_init\_\_.pyi.
I want (see #1632) the property types and default values to be injected automatically in the function signature.
The 'optional' syntax forces to import 'types' and is meant to disappear.

Now.
If it does not work, well it does not. So I have to step back at this point.
That will be part of the IDE autocompletion issue.
Thanks.

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**